doc/integrator/extend_application.rst

```diff
@@ -40,17 +40,243 @@ In the ``vars.yaml`` file your interface should be declared like that:
+
+- ``request`` -> the Pyramid request.
+- ``header`` -> the header additional part of the page, the ``dynamicUrl`` and ``interface`` meta, and the CSS inclusion.
+- ``spinner`` -> the spinner SVG image content.
+- ``footer`` -> the footer additional part of the page, for the JavaScript inclusion.
+
 Custom docker-compose service
 -----------------------------
 
-Will be filled later.
+In this chapter we will create a new Pyramid application that use cornice in a Docker image.
+
+We will use the Pyramid Cookiecutter starter, we can use it directly
+(by running ``cookiecutter gh:Pylons/pyramid-cookiecutter-starter``) if you want to do your
+image by your own, but in this tutorial we will get the files directly from the demo.
+For that run the following command (should be adapted for Windows but you will get the sense):
+
+.. prompt:: bash
+
+   cd /tmp
+   git clone git@github.com:camptocamp/demo_geomapfish.git
+   cd -
+   cp --recursive /tmp/demo_geomapfish/custom /tmp/demo_geomapfish/haproxy .
+
+Add in ``.prettierignore`` the following line:
+
+.. code::
+
+   custom/Pipfile.lock
+
+Apply the following diff in the ``setup.cfg``:
+
+.. code:: diff
+
+   - known_first_party=c2cgeoportal_commons,c2cgeoportal_geoportal,c2cgeoportal_admin,geomapfish_geoportal
+   + known_third_party=webtest
+   + known_first_party=c2cgeoportal_commons,c2cgeoportal_geoportal,c2cgeoportal_admin,geomapfish_geoportal,custom
+
+Add the following service in the ``docker-compose.yaml``:
+
+.. code:: yaml
+
+  custom:
+    image: ${DOCKER_BASE}-custom:${DOCKER_TAG}
+    build:
+      context: custom
+      args:
+        GIT_HASH: ${GIT_HASH}
+    environment:
+      - GUNICORN_CMD_ARGS=${GUNICORN_PARAMS}
+      - VISIBLE_WEB_HOST
+
+Add the following service in the ``docker-compose.override.sample.yaml``:
+
+.. code:: yaml
+
+  custom:
+    command:
+      - /usr/local/bin/gunicorn
+      - --reload
+      - --paste
+      - development.ini
+    volumes:
+      - ./custom/custom:/app/custom
+
+.. note::
+
+   If you needs the user credentials, you can do:
+
+   .. code:: python
+
+      requests.get(
+          "http://geoportal:8080/loginuser",
+          headers={"Cookie": request.headers.get("Cookie"), "Referer": request.referrer},
+      ).json()
 
 Custom JavaScript file
 ----------------------
 
-Will be filled later.
+In this example we will add a button in the tools bar, that open a new tool panel, that can be used to send a feedback.
+
+The tool button should be an instance of
+`gmfapi.elements.ToolButtonElement<https://camptocamp.github.io/ngeo/|main_branch|/apidoc/classes/srcapi_elements_ToolButtonElement.default.html>`_.
+
+In this example we will directly use
+`gmf-tool-button<https://camptocamp.github.io/ngeo/|main_branch|/apidoc/classes/srcapi_elements_ToolButtonElement.ToolButtonDefault.html>`_.
+
+Then we will include the following HTML in the canvas element, in ``geoportal/interfaces/desktop.html.mako``:
```

Comment:
This is part of "adding component to interface desktop"

Logical steps would be : 
- create service
- create / register interface
- create custom webcomponent
- add it to interface desktop